internal/store/migrations.go

```diff
@@ -578,6 +578,68 @@ var migrations = []migration{
 			}
 		}
 
+		return nil
+	}},
+	{semver.MustParse("0.11.0"), semver.MustParse("0.12.0"), func(e execer) error {
+		// Add version columns for all of the utilities.
+		_, err := e.Exec(`ALTER TABLE Cluster RENAME TO ClusterTemp;`)
+		if err != nil {
+			return err
+		}
+
+		_, err = e.Exec(`
+				CREATE TABLE Cluster (
+					ID                   TEXT PRIMARY KEY,
+					Provider             TEXT NOT NULL,
+					Provisioner          TEXT NOT NULL,
+					ProviderMetadata     BYTEA NULL,
+					ProvisionerMetadata  BYTEA NULL,
+					Version              TEXT NOT NULL,
+					Size                 TEXT NOT NULL,
+					State                TEXT NOT NULL,
+					AllowInstallations   BOOLEAN NOT NULL,
+					CreateAt             BIGINT NOT NULL,
+					DeleteAt             BIGINT NOT NULL,
+					LockAcquiredBy       TEXT NULL,
+					LockAcquiredAt       BIGINT NOT NULL,
+					PrometheusVersion    TEXT NOT NULL,
+					FluentbitVersion     TEXT NOT NULL,
+					NginxVersion         TEXT NOT NULL
```

Comment:
I am not entirely sure we want separate database entries for these. Something like the Metadata entries we do above may make more sense as they are flexible for multiple types of data about a variety of utilities.

In the future, we may have more/less utilities. Having each one have a hardcoded "presence" in the database and various models will probably be messy and/or a hassle in the future.

Reply:
I'm fine with following the pattern laid out by the Metadata entries at least for now, but after a lot of thinking and going back and forth with myself, I think we might want to discuss our overall strategy for storing "schemaless" data (my guess is that this will almost always be JSON). 

The current pattern we follow for mapping our database to an in-memory object is a little bit of a leaky abstraction, and there are a number of ways to address that but I want to avoid derailing myself. But that leaky abstraction will have runtime performance related effects in this PR: once I change the type in Cluster from 3 `string` fields to one `[]byte`, then anything that needs to operate on any one field in the JSON hidden in that `[]byte` will have to first unmarshal the blob, and then that unmarshalled struct also cannot be cached anywhere standard, since the Cluster field types need to map to the SQL column types (without a large change to the `store` package).

There are a number of approaches to overcome these limitations but they are all tradeoffs so we'll have to discuss. 

For now I'll just use `[]byte` in the struct and BYTEA in the database as types for the data and we will incur the (un)marshalling runtime costs as a tradeoff for a simplified database schema.